espnet2/train/preprocessor.py

```diff
@@ -258,6 +264,13 @@ def _convolve_rir(self, speech, power, rirs, tgt_fs=None, single_channel=False):
         rir = None
         if rir_path is not None:
             rir, fs = soundfile.read(rir_path, dtype=np.float64, always_2d=True)
+
+            if single_channel:
+                num_ch = rir.shape[1]
+                chs = np.random.permutation(num_ch).tolist()
+                rir = rir[:, chs]
```

Comment:
If my understanding is correct, `rir` is just shuffled along channel dim but still multi-channel. It might be better to make `rir` single-channel here since the rir convolution becomes redundant.

Reply:
You're right. I will update the code to only keep one channel.